Tasveer/Scenes/AllCollections/AllCollectionsViewModel.swift

```diff
@@ -11,12 +11,27 @@ import Foundation
 final class AllCollectionsViewModel {
     private let queue = OperationQueue()
     
-    func startUploadIfNeeded(for objects: [Collection?]) {
+    func fetchNewPhotosIfNeeded(for objects: [Collection?]) {
         let collections = objects.compactMap{ $0 }
-        for obj in collections where obj.syncStateValue == .none || (obj.syncStateValue == .syncing && obj.task?.isPaused == true) {
-            let operation = StartUploadingCollectionOperation(with: obj)
-            queue.addOperation(operation)
+        
+        let fetch = FetchNewPhotosForAllCollectionsOperation(withCollections: collections)
+        let upload = StartUploadingCollectionsOperation(withCollections: collections)
+        upload.addDependency(fetch)
+        
+        fetch.addCompletionBlock {
+            debugPrint("Fetch is completed")
         }
+        
+        queue.addOperation(fetch)
+        queue.addOperation(upload)
+    }
+    
+    func startUploadIfNeeded(for objects: [Collection?]) {
+        guard queue.operationCount <= 0 else { return }
```

Comment:
validating if current queue is empty. To avoid conflicting the operation for fetching new photos and start uploading new photos

Reply:
is this a general queue? - just that we make sure this is still valid even if we add other operations to the queue.

Reply:
This is a background queue which holds operations for All Collections scene. Currently it can holds operations that Fetch new photos on launch and start uploading unsynced photos